applications/IgaApplication/tests/cpp_tests/test_creation_utility.h

```diff
@@ -130,6 +131,52 @@ namespace TestCreationUtility
         return result_geometries(0);
     }
 
+
+    inline typename Geometry<NodeType>::Pointer GetQuadraturePointGeometryOnCurve(
+        ModelPart& rModelPart, SizeType PolynomialDegree, IntegrationPoint<3> IntegrationPoint)
+    {
+        typedef typename GeometryType::IntegrationPointsArrayType IntegrationPointsArrayType;
+        IntegrationPointsArrayType integration_points(1);
+        integration_points[0] = IntegrationPoint;
+        typedef typename Geometry<NodeType>::GeometriesArrayType GeometriesArrayType;
+
+        GeometriesArrayType result_geometries;
+
+        // Create the BrepCurve in Surface
+        // Assign the points belonging to the curve
+        PointerVector<Node> points_curve;
+        points_curve.push_back(Node::Pointer(new Node(1, 0.0, 0.05)));
+        points_curve.push_back(Node::Pointer(new Node(2, 1.0, 0.05)));
+        // Assign the curve's knot vector
+        Vector knot_vector_curve = ZeroVector(4);
+        knot_vector_curve[0] = 0.0;
+        knot_vector_curve[1] = 0.0;
+        knot_vector_curve[2] = 1.0;
+        knot_vector_curve[3] = 1.0;
+        // Polynomial degree of the curve
+        int p_curve = 1;
+        // Create the 2D embedded curve
+        auto curve = Kratos::make_shared<NurbsCurveGeometry<2, PointerVector<Node>>>(points_curve, p_curve, knot_vector_curve);
+
+        auto p_nurbs_surface = GenerateNurbsSurface(rModelPart, 3);
+        p_nurbs_surface->SetId(1);
+        IntegrationInfo integration_info_surface = p_nurbs_surface->GetDefaultIntegrationInfo();
+        p_nurbs_surface->CreateQuadraturePointGeometries(
+                result_geometries, 3, integration_points, integration_info_surface);
+
+        auto p_curve_nurbs_surface = Kratos::make_shared<NurbsCurveOnSurfaceGeometry<3, PointerVector<NodeType>, PointerVector<NodeType>>>(p_nurbs_surface, curve);
+
+        p_curve_nurbs_surface->SetId(2);
+        IntegrationInfo integration_info = p_curve_nurbs_surface->GetDefaultIntegrationInfo();
+        
+        p_curve_nurbs_surface->CreateQuadraturePointGeometries(result_geometries, 3, integration_points, integration_info);
+
+        rModelPart.AddGeometry(p_nurbs_surface);
+        rModelPart.AddGeometry(p_curve_nurbs_surface);
+
+        return result_geometries(0);
```

Comment:
The result container must be passed by reference as an input of the function. Indeed, it compiles and works because you're using a pointer, which is copied when assigning the value.